src/main/java/dev/snowdrop/release/services/GitService.java

```diff
@@ -191,23 +215,28 @@ private GitConfig(String org, String repo, String branch) {
             }).thenApplyAsync(branches -> branches.stream().noneMatch(ref -> ref.getName().contains(branch)));
         }
 
-        public static GitConfig githubConfig(String gitRef, String token) {
+        public static GitConfig githubConfig(String gitRef, String token, Optional<String> cloneFromGitRef) {
             final var split = gitRef.split("/");
             if (split.length != 3) {
                 throw new IllegalArgumentException("Invalid git reference: " + gitRef
                     + ". Must follow organization/repository/branch format.");
             }
-            return new GitHubConfig(split[0], split[1], split[2], token);
+            return new GitHubConfig(split[0], split[1], split[2], token, cloneFromGitRef);
+        }
+
+        public static String getDetaultBranchName(final String release) {
```

Comment:
This method should be used wherever it's needed instead of copying the same logic several times.